secretkeylabs/xverse-web-extension
# Fix Tx history screen reload & send form navigation issues

# 🔘 PR Type
What kind of change does this PR introduce?




# 📜 Background
Transaction history page for all transaction reloads every couple of seconds showing a spinner every time which is not the best UX. 
There were also navigation issues while fund transfer. 
On pressing the back button on the STX send form goes back to the confirmation screen, which is incorrect behavior.
The input fields also get cleared when user navigates back from confirmation screen to send screen

Issue Link: #[[ENG-2678](https://linear.app/xverseapp/issue/ENG-2678/navigation-issue-when-sending-stx)]
              [ENG-2588](https://linear.app/xverseapp/issue/ENG-2588/transaction-history-page-for-all-transaction-reloads-every-couple-of)

# 🔄 Changes

- The transaction history UX is improved so that the spinner is not shown on refetch also the screen does not scroll up every time the reloading happens. 
- The navigation issue in the STX flow is fixed where  the user clicks the back button on the send screen you get redirected to the home page
- The clearance of input fields when the user navigates back was introduced when the check was added to clear funds transfer progress if the account is switched. The useEffect was being called on the first render hence the input fields would get cleared if the user navigates backward from the confirmation screen to the send screen. The PR fixes the bug by creating a customer hook `useClearFormOnAccountSwitch`. The hook sees if its the first render then doesn't clear the input fields. 


# 🖼 Screenshot / 📹 Video

## Transaction history refetch improvement 

### Before

https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/f03a9410-98d2-4593-962b-4c4d732badeb

### After

https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/27ab8aea-6801-48d2-89b7-1f0c87d7a3f0

## Navigation issues

https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/8e3ec365-0667-4d98-beea-cba0abb83ef8

## Change account flow 

https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/c38ab77d-0689-41ed-9d60-e090511c23cc


# 🧪 Testing Instructions 
- Verify the transaction history screen reload issue is fixed where the spinner and jump to top does not happen after every couple of seconds 
- Try the STX/BTC send flow and see if you go back from the confirm screen to send screen the input fields are populated.
- On pressing the back button on the STX send form goes back to the main dashboard
- Also confirm the flow where if you are in the middle of filling the form and the account is changed then the expected behavior should be that the input fields should be cleared. 


# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.
